### PR TITLE
Fix FreeJoint world Jacobian translation

### LINKED ISSUE #1657: dart seems to return incorrect world jacobian for the tranlsation degrees of freedom of a freejoint
### Bug Report
- [x] I checked the [documentation](http://dartsim.github.io/) and the [forum](https://dartsim.discourse.group/) but found no answer.
- [x] I checked to make sure that this issue has not already been filed.

#### Environment
* DART version: 6.12.1
* OS name and version name(or number): Ubuntu 18.04 and newer
* Compiler name and version number: GCC 8.0.0 or newer
* Using dartpy python API 

#### Expected Behavior
dart::dynamics::bodyNode::getJacobian(offset, dart::dynamcis::Frame::World) or dart::dynamics::bodyNode::getWorldJacobian(offset) returning a jacobian in world coordinates also when using a dart::dynamics::freejoint as first joint.

#### Current Behavior
given a dart::dynamics::freejoint the jacobian entries corresponding to the translational degrees of freedom of the joint (q[3:6]) seem to be incorrect.

#### Steps to Reproduce
1. create a bodyNode with a freeJoint
2. set the a rotational degree of freedom of the joint, such that the body is not well aligned with the world frame (e.g. q[0] = 1)
3. obtain the worldJacobian for the center of the bodyNode and observe the jacobian entries corresponding to the translational degrees of freedom of the joint (e.g worldJacobian[(3:6),(3:6)])
4. observe that the entries of the jacobian are not an identity matrix. 
5. observe that when the translational degrees of freedom q[3:6] are set , the body translates along the x,y,z coordinates of the world frame.
6. since the worldJacobian relates the positional changes of the body given in the world frame to the changes of the generalized coordinates, the jacobian entries for these translational should be an identity matrix.

#### Code to Reproduce
A minimal example to reproduce the observed behaviour is attached as python code. (just paste code provided in the attached .txt file in a .py file and execute with python 3)
It provides further explanation and visualization of what behavior is observed.

Further a little workaround which worked for my case is presented.
Solved by multiplying the part of the returned jacobian matrix which corresponds to the translational degrees of freedom with the inverse of the relative Transform's orientation of the freejoint. 
Unfortunately, I am not able to provide a solution or fix for the C++ implementation.

[testDartWorldJacobian.txt](https://github.com/dartsim/dart/files/8253389/testDartWorldJacobian.txt)

### PR DESCRIPTION
**Pull request by jslee02**
Fixes #1657.

This updates `dynamics::FreeJoint` Jacobian/integration logic to remain consistent when `T_parentBodyToJoint` and `T_childBodyToJoint` are non-identity.

- Fix `FreeJoint::integratePositions()` to integrate the relative transform (body twist) and solve back for `Q`, preserving correct motion with offset joint frames.
- Fix `FreeJoint::getRelativeJacobianStatic(positions)` to use the passed positions (instead of a cached Jacobian).
- Ensure `setRelativeSpatialVelocity()` uses the up-to-date pose-dependent Jacobian; fix `updateRelativeJacobianTimeDeriv()` for non-identity child joint frames.
- Add C++ integration coverage: FD world Jacobian + classic-deriv checks (including near-π), pose-change setter regression, and energy conservation checks.
- Add GUI example `free_joint_cases` (ImGui) to visualize linear-only/angular-only/both and edge cases, with built-in numeric checks.

---

#### Before creating a pull request

- [ ] Run `pixi run test-all` to lint, build, and test your changes (CI passed; locally ran `pixi run lint`, `pixi run test`, `pixi run test-py`)
- [x] Add unit tests for new functionality
- [ ] Document new methods and classes
- [ ] Add Python bindings (dartpy) if applicable

### CODE REVIEW

**Comment by jslee02** [comment]
@codex review

**Comment by jslee02** [comment]
@codex review

File: dart/dynamics/FreeJoint.cpp [hunk 1]
```diff
@@ -485,19 +483,38 @@ void FreeJoint::setAngularAcceleration(
...
-  const Eigen::Isometry3d T2 = convertToTransform(_q2);
+  const Eigen::Isometry3d Q1 = convertToTransform(_q1);
+  const Eigen::Isometry3d Q2 = convertToTransform(_q2);
+
+  Eigen::Vector6d diff = Eigen::Vector6d::Zero();
+  const Eigen::Matrix3d rotation1 = Q1.linear();
+  const Eigen::Matrix3d rotationChange = rotation1.transpose() * Q2.linear();
+
+  diff.head<3>() = rotation1 * math::logMap(rotationChange);
+  diff.tail<3>() = Q2.translation() - Q1.translation();
```
[the remark this answers is not in the document]
Comment (jslee02) [reply dart/dynamics/FreeJoint.cpp]: Thanks for calling this out. For FreeJoint in this PR the generalized velocities are world-frame (omega, v) (this is what makes the world Jacobian translational block identity, per #1657), so getPositionDifferences() is intentionally left-trivialized / world-frame: for small dt, getPositionDifferences(q(t+dt), q(t))/dt matches qdot. This is also required by INTEGRATION_dynamics_DynamicsFiniteDifference. I added/updated tests to make the world-frame convention explicit and to cover the spherical-inertia constant-twist long-horizon case.
